junkyard-styled-innersourcing.md

```diff
@@ -0,0 +1,35 @@
+## Title
+* Junkyard styled inner sourcing
+* Finding but deciding not to use the internal component
+
+## Context
+Two situations:
+
+* SW component not designed for reuse because the schedule was too tight and did not allow for the extra effort needed
+* Did not have the knowledge or training to create components that can b reused.
+
+## Problem
+* How do you encourage people to share whatever code they write?
+
+## Forces
+* People don't share something so others can use it; they only share it for the sake of sharing. They wouldn't mind if no one uses it (a bit of a graveyard: "InnerSource is where modules go to die")
+* A developer might be a person who likes to share their code; one that likes others to read it (and maybe comment on it. But mostly they like to write code. "Hey, I wrote this nifty gzorp and you might like it (but its no skin off my nose if you don't)"
+* the code is crap
+* the code doesn't address any of the company's problems (at the moment anyway)
+* How does the person let others know that the code exists?
+* We want the person to contribute code, so what roadblocks need to be removed so that they continue making their code available?
+* How do we politely take their freely offered code and harden it or otherwise prepare it for production without having the original contributor feel like we are saying their code is crap?
+
```

Comment:
Suggestion for additional force: _The more responsive the original author is to requests for changes coming from potential re-users of her code, the more likely it is that it will be reused._